### PR TITLE
Fixed the media fail error on caller's side

### PR DESCRIPTION
**Pull request by DantrazTrev**
Fixes vector-im/element-web/issues/16747

Switched `CallType` before async function calls  in `/src/webrtc/call.ts` that were causing the `showMediaCaptureError` in `matrix-react-sdk/src/CallHandler.tsx` to skip all the conditions and hence causing a generic error.

Signed-off-by: Ayush Pratap Singh ayushpratap16@gmail.com

### CODE REVIEW

**Comment by t3chguy** [comment]
Needs DCO sign-off as per https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.rst#sign-off

File: src/webrtc/call.ts [hunk 1]
```diff
@@ -394,6 +396,7 @@ export class MatrixCall extends EventEmitter {
 
             logger.debug("Got screen stream, requesting audio stream...");
             const audioConstraints = getUserMediaContraints(ConstraintsType.Audio);
+            this.type = CallType.Video;
```
Comment (SimonBrandner) [inline src/webrtc/call.ts:399]: Is there any reason to put this inside and not before the try-catch block? I am probably just missing something here :sweat_smile:

Comment (DantrazTrev) [reply src/webrtc/call.ts:399]: Oh no , sorry I would just fix it

**Comment by DantrazTrev** [comment]
> Needs DCO sign-off as per https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.rst#sign-off

Done

File: src/webrtc/call.ts (hunk 1 above)
Comment (SimonBrandner) [reply src/webrtc/call.ts]: Hmm, I've just realized this might actually do something we want by accident :D

Comment (DantrazTrev) [reply src/webrtc/call.ts]: Oh, just a sec let me just check . I think it would work the either way

Comment (SimonBrandner) [reply src/webrtc/call.ts]: I think an intelligent thing to do here would be to decouple the call type and the `NoUserMedia` error. But I wouldn't worry about it if you don't want to

Comment (SimonBrandner) [reply src/webrtc/call.ts]: The current solution will work fine

Comment (DantrazTrev) [reply src/webrtc/call.ts]: I could do that but I've a naive question how do you make screen share calls here? I couldn't figure out

Comment (SimonBrandner) [reply src/webrtc/call.ts]: Screen-share calls are a bit of a hack right now... You can shift-click the video call button to initiate a screen-sharing call

Comment (DantrazTrev) [reply src/webrtc/call.ts]: > The current solution will work fine

Alright

**Review by SimonBrandner — APPROVED** [review]
LGTM, just fix the unnecessary new lines

**Review by turt2live — APPROVED** [review]
Seems plausible, thanks!

**Pull request merged**